tools/tools.go

```diff
@@ -338,22 +341,107 @@ func EnvmanClear(envstorePth string) error {
 	return nil
 }
 
-// EnvmanRun ...
+const enableSecretFilteringKey = "BITRISE_ENABLE_SECRET_FILTERING"
+
+// filteringEnabled returns true if enableSecretFilteringKey env presents with true value, false otherwise.
+func filteringEnabled(secrets []envmanModels.EnvironmentItemModel) bool {
+	for _, secret := range secrets {
+		key, value, err := secret.GetKeyValuePair()
+		if err != nil {
+			return false
+		}
+		if key == enableSecretFilteringKey && value == "true" {
+			return true
+		}
+	}
+	return false
+}
+
+// EnvmanRun runs a command through envman.
 func EnvmanRun(envstorePth, workDirPth string, cmdArgs []string, timeout time.Duration, secrets []envmanModels.EnvironmentItemModel) (int, error) {
 	logLevel := log.GetLevel().String()
 	args := []string{"--loglevel", logLevel, "--path", envstorePth, "run"}
 	args = append(args, cmdArgs...)
 
+	if !filteringEnabled(secrets) {
+		command := command.NewWithStandardOuts("envman", args...).SetStdin(os.Stdin).SetDir(workDirPth)
+
+		if timeout <= 0 {
+			exitCode, err := command.RunAndReturnExitCode()
+			if err != nil {
+				err = errors.WithStack(err)
+			}
+			return exitCode, err
+		}
+
+		// create a new process group for our process and its child processes
+		cmd := command.GetCmd()
+		cmd.SysProcAttr = &syscall.SysProcAttr{Setpgid: true}
+
+		if err := cmd.Start(); err != nil {
+			return -1, errors.WithStack(err)
+		}
+
+		// Setpgid: true creates a new process group for cmd and its subprocesses
+		// this way cmd will not belong to its parent process group,
+		// cmd will not be killed when you hit ^C in your terminal
+		// to fix this, we listen and handle Interrupt signal manually
+		c := make(chan os.Signal, 1)
+		signal.Notify(c, os.Interrupt)
+		go func() {
+			<-c
+			if err := syscall.Kill(-cmd.Process.Pid, syscall.SIGKILL); err != nil {
+				log.Warnf("Failed to kill process, error: %+v", errors.WithStack(err))
+			}
+			os.Exit(130)
+		}()
+		defer func() {
+			signal.Stop(c)
+		}()
+		//
+
+		timer := time.AfterFunc(timeout, func() {
+			if err := syscall.Kill(-cmd.Process.Pid, syscall.SIGKILL); err != nil {
+				log.Warnf("Failed to kill process, error: %+v", errors.WithStack(err))
+			}
+		})
+
+		err := cmd.Wait()
+
+		timer.Stop()
+
+		if err != nil {
+			if err.Error() == "signal: killed" {
```

Comment:
please do not use hardcoded error messages, or use them as const with the prefix of err like `errSigKill`.